src/main/java/org/opensearch/ad/rest/RestGetAnomalyDetectorAction.java

```diff
@@ -112,7 +113,34 @@ protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOExc
     public List<ReplacedRouteHandler> replacedRouteHandlers() {
         String path = String.format(Locale.ROOT, "%s/{%s}", AnomalyDetectorExtension.LEGACY_OPENDISTRO_AD_BASE_URI, DETECTOR_ID);
         String newPath = String.format(Locale.ROOT, "%s/{%s}", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, DETECTOR_ID);
-        return ImmutableList.of(new ReplacedRouteHandler(RestRequest.Method.GET, newPath, RestRequest.Method.GET, path, handleRequest));
+        return ImmutableList
+            .of(
+                new ReplacedRouteHandler(RestRequest.Method.GET, newPath, RestRequest.Method.GET, path, handleRequest),
+                new ReplacedRouteHandler(RestRequest.Method.HEAD, newPath, RestRequest.Method.HEAD, path, handleRequest),
```

Comment:
I see the above line replicates existing functionality, but these next 3 handlers look new.  Do they already exist in another `Rest<something>Action` somewhere?  If so, why aren't we adding these route(handler)s there and not here?

Reply:
I just tried to immitate how it is for RestGetAnomalyDetectorAction in Plugins. These handlers were part of RestGetAnomalyDetectorAction for  **public List<ReplacedRoute> replacedRoutes()** for plugins. You can find them in this same class in the commented code below.

Reply:
Ah, there it is! It wasn't part of the diff so I didn't see it.

Can we just uncomment the replaced routes code below and put this code there?

Reply:
Also I see there are two routes commented out, are those still TODO or does this code handle them?

Reply:
Those two routes will be covered in https://github.com/opensearch-project/opensearch-sdk-java/issues/451